### PR TITLE
fix(web): Catch battery status promise rejections

### PR DESCRIPTION
**Pull request by bradjones1**
## Description

Fixes https://github.com/react-native-device-info/react-native-device-info/issues/1544

## Compatibility

Web-only fix.

## Checklist

* [x] I have tested this on a device/simulator for each compatible OS
* `[n/a]` I added the documentation in `README.md`
* `[n/a]` I updated the typings files (`privateTypes.ts`, `types.ts`)
* `[n/a]` I added a sample use of the API (`example/App.js`)

### CODE REVIEW

**Comment by machour** [comment]
LGTM, but should this be considered as a BC breaking change? 
Some users may be try/catching and handling errors differently than what you're doing here..

**Comment by bradjones1** [comment]
>  Some users may be try/catching and handling errors differently than what you're doing here..

This is a fair point - how about a more narrow change that only affects the `init()` call, since that basically can't be caught right now (which is why I worked on this to begin with)?

**Comment by bradjones1** [comment]
Frustratingly, it doesn't appear that the `catch()` at https://github.com/react-native-device-info/react-native-device-info/pull/1545/files#diff-a2eea7d51d5408108433b3aa81ed471741b8f15ae58ca9dd358da064f819c3b5R84 is consistently working on Windows. I have seen this resurface recently, although with much less frequency. Chrome 86 and 100. I can't recreate it with a similar browser setup on LambdaTest... invoking `navigator.getBattery()` in the JS console returns the `BatteryManager`.

**Comment by mikehardy** [comment]
Thanks for your patience - sorry my time here is so intermittent, but I can't promise more in general 😬 - at any rate, I like the idea of the catches, but if it is not consistent it doesn't help much

(though as an aside: what !? how can a catch not consistently catch...how is that even a thing? So difficult trying to produce reliable code for all platforms...)

As a policy note on breaking changes (assuming consistent behavior can be created...) I am okay with breaking changes in general, my stance is "version numbers are free". My requirement is more about developer empathy in that breaking changes *do* get a semver major, *and* they are accompanied with a really good developer note explaining exactly who is affected, and exactly what changed, and a reasonable attempt to specify the likely change for adoption. That's all - and it's not difficult + and I usually handle that myself during the commit so that the auto-release infrastructure picks it up.

TL;DR: no big deal to do a breaking change normally

**Comment by mikehardy** [comment]
Marking as "on hold" until behavior consistency is achieved so whatever path forward is the final path forward is reliable vs flaky
